### DISCUSSION TITLE
Syntax highlighting not working as promised

### DISCUSSION THREAD

**Discussion by dedmen** [community/community#21867] [Other Feature Feedback, Questions, & Ideas]
<p><a href="https://blog.github.com/2016-02-01-github-pages-now-faster-and-simpler-with-jekyll-3-0/#the-highlight-zone" rel="nofollow noopener">Everywhere</a> I <a href="https://help.github.com/articles/using-syntax-highlighting-on-github-pages/" rel="nofollow noopener">look</a> it says that “Syntax highlighting on GitHub Pages works the same way it does on GitHub” which is true for some languages but not for all.</p>
<p>For example here: <a href="https://github.com/arma3/arma3.github.io/blob/master/index.md" rel="nofollow noopener">https://github.com/arma3/arma3.github.io/blob/master/index.md</a><br>
All the examples work as expected. Note here the SQF highlight that correctly highlights the SQF command (getPos) vs the variable (SQFplayer).</p>
<p>However if you look at the final page <a href="http://sqf.ovh/" rel="nofollow noopener">http://sqf.ovh/</a> (we use a special stylesheet for JS syntax. But it cannot differentiate between variable and command)</p>
<p>You’ll see that JS and CPP highlights are working in that they are highlighting something. However the SQF highlight is just blank.</p>
<p>Github says syntax highlighting works the same as on Github itself, but it doesn’t seem to be like that.</p>
<p>Any advice on what I should do? Do I need to add the language to Rogue and Pygments manually although Github already says it should work?</p>
<p>I just found <a href="https://github.com/jneen/rouge/pull/761" rel="nofollow noopener">https://github.com/jneen/rouge/pull/761</a> but it still seems wrong that Github says it should work the same while it actually doesn’t.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You’re right that the documentation could probably be a little more specific but what it means when it says “syntax highlighting works the same as on GitHub” is that you can use fenced code blocks in the same way on GitHub Pages as you can on GitHub. The syntax highlighting itself relies on the Rouge library and, as you pointed out, Rouge itself does not yet support SQF.</p>
<p>Sorry for the confusion. I’ve given the feedback to the documentation team.</p>
